AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/SQLiteResultError.swift

```diff
@@ -0,0 +1,28 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import SQLite
+import Amplify
+
+/// Checks for specific SQLLite error codes
+/// See https://sqlite.org/rescode.html#primary_result_code_list for more details
+struct SQLiteResultError {
```

Comment:
have you considered representing `SQLiteResultError` with an enum? And we can have a factory method to bridge a `DataStoreError` with it?

Reply:
example
```
extension DataStoreError
  static func toSQLiteError() -> SQLiteError? 
}

enum SQLiteError {
  case constraintViolation
  case other(message, code)

  init(error: DataStoreError) throw {
  }
}
```